tree.rb

```diff
@@ -25,5 +27,14 @@ def add_child(value)
   end
 
   def each(&block)
+    node = self.children
+    until node.empty?
+      block.call(node.head.value)
+      node = node.children
+    end
```

Comment:
Actually...  this code existed already.  I didn't write this code.  so... no.. i don't think it works.  :)  actually, I'm glad you asked that.  it's why I didn't use Tree.each in my Trie.    anyway...   hi